state-chain/pallets/cf-validator/src/lib.rs

```diff
@@ -765,6 +767,33 @@ pub mod pallet {
 			}
 			T::AccountRoleRegistry::register_as_validator(&account_id)
 		}
+
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::ValidatorWeightInfo::deregister_as_validator())]
+		pub fn deregister_as_validator(origin: OriginFor<T>) -> DispatchResult {
+			let account_id = T::AccountRoleRegistry::ensure_validator(origin.clone())?;
+
+			let validator_id = <ValidatorIdOf<T> as IsType<
+				<T as frame_system::Config>::AccountId,
+			>>::from_ref(&account_id);
+
+			ensure!(
+				(LastExpiredEpoch::<T>::get()..=CurrentEpoch::<T>::get())
+					.all(|epoch| !HistoricalAuthorities::<T>::get(epoch).contains(validator_id)),
+				Error::<T>::StillKeyHolder
+			);
+
+			<pallet_session::Pallet<T>>::purge_keys(origin)?;
+
+			if let Some((peer_id, _, _)) = AccountPeerMapping::<T>::take(&account_id) {
+				MappedPeers::<T>::remove(peer_id);
+				T::CfePeerRegistration::peer_deregistered(validator_id.clone(), peer_id);
+			}
+
+			T::AccountRoleRegistry::deregister_as_validator(&account_id)?;
```

Comment:
No - if they can withdraw all their funds, whether bidding or not, doesn't really matter. Of course in order to no longer be in the active set, it's likely they are no longer bidding.

Reply:
But consider the scenario where they are not an active keyholder, they've increased their bid such that win the auction, and now they call deregister as validator. This would still be callable but it shouldn't be after they've entered the auction.

Reply:
OK I see, yeah I suppose that could be abused... (side note: we should move bidding/not bidding to the validator pallet...)